From 7a465c107e65d28dd146e48953e9695f9bfea483 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Tue, 5 Mar 2019 09:10:57 +0100
Subject: [PATCH 9/9] [DNM] Squash of lineage-16.0-android-9.0.0_r34

Squashed commit of the following:

commit 20ac1203a3201ac3e6d05a19325f5569033f3d08
Author: Martijn Coenen <maco@google.com>
Date:   Wed Dec 5 09:42:25 2018 +0100

    Export maximum number of fds/ints in a native_handle.

    So we can deserialize it consisently and safely.

    Bug: 120084106
    Test: builds
    Change-Id: I0eafff70d3a7e4d732fe600a0052efb90108208d

Change-Id: Ife58ec159652c27590e7888abd57642dff090bc2
---
 libcutils/include/cutils/native_handle.h | 3 +++
 libcutils/native_handle.cpp              | 6 ++----
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/libcutils/include/cutils/native_handle.h b/libcutils/include/cutils/native_handle.h
index 10f5bc094..f6cae360f 100644
--- a/libcutils/include/cutils/native_handle.h
+++ b/libcutils/include/cutils/native_handle.h
@@ -23,6 +23,9 @@
 extern "C" {
 #endif
 
+#define NATIVE_HANDLE_MAX_FDS 1024
+#define NATIVE_HANDLE_MAX_INTS 1024
+
 /* Declare a char array for use with native_handle_init */
 #define NATIVE_HANDLE_DECLARE_STORAGE(name, maxFds, maxInts) \
     alignas(native_handle_t) char (name)[                            \
diff --git a/libcutils/native_handle.cpp b/libcutils/native_handle.cpp
index 66f7a3d59..b409e5b9d 100644
--- a/libcutils/native_handle.cpp
+++ b/libcutils/native_handle.cpp
@@ -22,9 +22,6 @@
 #include <string.h>
 #include <unistd.h>
 
-static const int kMaxNativeFds = 1024;
-static const int kMaxNativeInts = 1024;
-
 native_handle_t* native_handle_init(char* storage, int numFds, int numInts) {
     if ((uintptr_t) storage % alignof(native_handle_t)) {
         errno = EINVAL;
@@ -39,7 +36,8 @@ native_handle_t* native_handle_init(char* storage, int numFds, int numInts) {
 }
 
 native_handle_t* native_handle_create(int numFds, int numInts) {
-    if (numFds < 0 || numInts < 0 || numFds > kMaxNativeFds || numInts > kMaxNativeInts) {
+    if (numFds < 0 || numInts < 0 || numFds > NATIVE_HANDLE_MAX_FDS ||
+        numInts > NATIVE_HANDLE_MAX_INTS) {
         errno = EINVAL;
         return NULL;
     }
-- 
2.17.1

